File: MyGame/app/src/main/java/com/example/mygame/domain/platform/MovingPlatformOnX.kt
package com.example.mygame.domain.platform

import com.example.mygame.domain.GameConstants
import com.example.mygame.domain.Screen

class MovingPlatformOnX(
    createdX: Float,
    createdY: Float
) : Platform(createdX, createdY) {
    private var directionX = directionXRandom()

    enum class DirectionX(val value: Int) {
        LEFT(0),
        RIGHT(1)
    }

    fun changeDirectionX(screen: Screen) {
        if (right >= screen.width) {
            directionX = DirectionX.LEFT
        } else if (left <= 0f) {
            directionX = DirectionX.RIGHT
        }
    }

    private fun directionXRandom() : DirectionX {
        val value = (DirectionX.LEFT.value..DirectionX.RIGHT.value).random()

        return if (value == DirectionX.LEFT.value) {
            DirectionX.LEFT
        } else {
            DirectionX.RIGHT
        }
    }

    override fun updatePosition(elapsedTime: Float) {
        if (directionX == DirectionX.LEFT) {
            x -= GameConstants.PLATFORM_ON_X_SPEED * elapsedTime
        } else {
            x += GameConstants.PLATFORM_ON_X_SPEED * elapsedTime
        }
    }
}

File: GameServer/src/Model/Platform/MovingPlatformOnX.php
<?php

namespace App\Model\Platform;

use App\Model\Screen;
use App\Model\Platform;
use App\Constants\JsonConstants;
use App\Constants\GameConstants;

class MovingPlatformOnX extends Platform
{
    public const DIRECTION_LEFT = 0;
    public const DIRECTION_RIGHT = 1;

    private $directionX;

    private $speedX;

    public function __construct($createdX, $createdY)
    {
        parent::__construct($createdX, $createdY);
        $this->directionX = $this->directionXRandom();
    }

    public function changeDirectionX(Screen $screen)
    {
        if ($this->getRight() >= $screen->right) {
            $this->directionX = self::DIRECTION_LEFT;
        } else if ($this->x <= $screen->left) {
            $this->directionX = self::DIRECTION_RIGHT;
        }
    }

    private function directionXRandom()
    {
        $value = rand(self::DIRECTION_LEFT, self::DIRECTION_RIGHT);
        return $value;
    }

    public function updatePosition($elapsedTime): void
    {
        if ($elapsedTime > 1.0) {
            return;
        }
        $this->speedX = GameConstants::PLATFORM_ON_X_SPEED * $elapsedTime;

        if ($this->directionX == self::DIRECTION_LEFT) {
            $this->x -= $this->speedX;
        } else {
            $this->x += $this->speedX;
        }
    }

    public function toArray(): array
    {
        return [
            JsonConstants::MOVING_PLATFORM_ON_X_TYPE,
            round($this->x, 2),
            round($this->y, 2),
            round($this->speedX, 2),
            0,
            0,
        ];
    }
}
